building-and-deploying-a-fastapi-app.md

```diff
@@ -0,0 +1,494 @@
+
+
+Before proceeding, create a _.gitignore_ file in the project folder to prevent checking in the "venv" folder to git:
+
+
+```
+$ touch .gitignore
+```
+
+Add the following:
+
+
+```
+venv
+__pycache__
+```
+
+With the necessary installations in place, log on to Okteto from your terminal using the command:
+
+```sh
+$ okteto login
+```
+
+The next step is to create a stack manifest file for Okteto. The manifest file removes the need to deal with the complexities of Kubernetes manifests.
+
+```
+$ touch okteto-stack.yaml
+```
+
+In the `okteto-stack.yaml` file, define the application using a format that's very similar to docker-compose:
+
+```yaml
+name: fastapi-crud
+services:
+  fastapi:
+    public: true
+    image: fastapi-crud:latest
+    build: .
+    replicas: 1
+    ports:
+      - 8080
+    resources:
+      cpu: 100m
+      memory: 128Mi
+```
+
+In the manifest file above, you defined your application's name and the services your app is made up of, fastapi. The fastapi service exposes your application via the port `8080` to a public HTTPs endpoint with a valid certificate. The service also allocates itself with 100 mili CPUs and 128Mi of memory. 
+
+> Okteto has a reference page on [https://okteto.com/docs/reference/stacks](Okteto Stacks Manifest).
+
+With the manifest in place, create a Dockerfile to house the build instruction for the application. Dockerfile contains build instruction for images to be deployed on Okteto.
+
+```sh
+$ touch Dockerfile
+```
+
+Add the following:
+
+```Dockerfile
+FROM python:3.8
+
+ADD requirements.txt /requirements.txt
+
+ADD main.py /main.py
+
+ADD okteto-stack.yaml /okteto-stack.yaml
+
+ADD okteto.yml /okteto.yml
+
+RUN pip install -r requirements.txt
+
+EXPOSE 8080
+
+COPY ./app app
+
+CMD ["python3", "main.py"]
+```
+
+Before proceeding to deployment on Okteto, download and set up your Kubernetes credentials on your local machine:
+
+![Okteto Dashboard](https://res.cloudinary.com/adeshina/image/upload/v1608635743/x9tmv4styid2hffxilxx.png)
+
+Let's deploy your application to Okteto. Start by running the command: 
+
+```
+okteto namespace
+```
+
+The command above setups a namespace on the Okteto cloud. Next, run the command to deploy the application:
+
+```
+okteto stack deploy --build
+```
+
+The command above builds the service you indicated in the `okteto-stack.yml` file and then deploys the application. This command eases the stress of having to build, then manually configuring the application after deployment.
+
+The command above starts the application. Navigate to your dashboard and click the link under the "Endpoints" heading:
+
+![Dashboard](https://res.cloudinary.com/adeshina/image/upload/v1608638743/q8xcjhl2gdnh0tdzr573.png).
+
+Go on and test the endpoints.
+
+You can exit the development console by running the Ctrl + D command twice.
```

Comment:
since we are not running `okteto up` this line is not needed.

Reply:
Ah, that's true!

Reply:
Fixed